tests/sentry/web/frontend/test_oauth_revoke.py

```diff
@@ -0,0 +1,245 @@
+from functools import cached_property
+from typing import List
+
+import pytest
+from django.urls import reverse
+
+from sentry.models import ApiApplication, ApiGrant, ApiToken
+from sentry.testutils.cases import TestCase
+from sentry.testutils.silo import control_silo_test
+
+
+@control_silo_test(stable=True)
+class OAuthRevokeTest(TestCase):
+    @cached_property
+    def path(self):
+        return reverse("sentry-oauth-revoke")
+
+    def setUp(self):
+        super().setUp()
+
+        # create a dummy api application
+        self.application = ApiApplication.objects.create(
+            owner=self.user,
+            redirect_uris="http://localhost:1234",
+        )
+
+        # create a second application to test app ownership boundary
+        self.other_application = ApiApplication.objects.create(
+            owner=self.user,
+            redirect_uris="http://localhost:5678",
+        )
+
+        # authorize the api application to act on behalf of the test user
+        self.grants: List[ApiGrant] = []
+
+        for i in range(4):
+            grant = ApiGrant.objects.create(
+                user=self.user,
+                application=self.application,
+                redirect_uri="http://localhost:1234",
+            )
+            self.grants.append(grant)
+
+        # create some oauth tokens
+        self.tokens: List[ApiToken] = []
+
+        for i in range(4):
```

Comment:
nit: you could condense these two loops into one (would use fewer lines of code and be just a tiny bit faster)

Reply:
Good idea! Thank you!